morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlDialect.java

```diff
@@ -4402,6 +4403,16 @@ protected String getSqlForInsertInto(@SuppressWarnings("unused") InsertStatement
   }
 
 
+  /**
+   * Returns any statements needed to automatically heal the given schema.
+   *
+   * @param schemaResource Schema resource that can be examined.
+   * @return List of statements to be run.
+   */
+  public List<String> getSchemaConsistencyStatements(SchemaResource schemaResource) {
+    schemaResource.getDatabaseMetaDataProvider();

```

Comment:
is this line just here as an example of how you might get the consistency statements?

Reply:
Yes. I ended up implementing a fully working solution example in Postgres,
but yes, this is where you start to get info out of your associated DatabaseMetaDataProvider.